Log aot and idx waitcounters.

jovianjaison opened the pull request:
Summary:
Added for create_aot_dispatcher_function and compile_fx_inner.

Note:
Log wait counters flag is already set for: 
1. async_compile.precompile
2. remote_fx_graph_cache_get
3. remote_fx_graph_cache_put

Test Plan: contbuild

Differential Revision: D73866124




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

masnesral commented on torch/_inductor/compile_fx.py:
```diff
@@ -1289,7 +1290,9 @@ def codegen_and_compile(
                     compiled_fn_runner = None
                     with dynamo_timed(
-                        "GraphLowering.compile_to_fn", log_pt2_compile_event=True
+                        "GraphLowering.compile_to_fn",
```
I'm actually skeptical of any waitcounter in a dynamo_timed() where we don't already have a dynamo_compile_column_us param specified (which causes logging to dynamo_compile scuba). I'd generally expect dynamo_compile to be finer-grained, so it indicates to me the waitcounter is probably not needed -or- we should really be adding a new dynamo_compile column as well.

masnesral commented on torch/_functorch/aot_autograd.py:
```diff
@@ -572,7 +572,12 @@ def create_aot_dispatcher_function(
     shape_env: Optional[ShapeEnv],
 ) -> tuple[Callable, ViewAndMutationMeta]:
-    with dynamo_timed("create_aot_dispatcher_function", log_pt2_compile_event=True):
+    with dynamo_timed(
```
Why is this important to have a waitcounter? I only ask because if it's "important" then I'd expect we should be logging this time to dynamo_compile as well?

jovianjaison replied:
We'll need this to measure time spent in aot. I'll stack a diff to add this to the dynamo_compile logger config and log it here then as well.

masnesral replied:
I really don't know the aot stuff very well so I'm taking for granted this is the right thing to time

masnesral commented on torch/_inductor/async_compile.py:
```diff
@@ -355,24 +355,28 @@ def get_result() -> CachingAutotuner:
-                "async_compile.precompile",
-                log_pt2_compile_event=True,
-                dynamo_compile_column_us="triton_compile_time_us",
+                "compile_async",
```
Let's not introduce another nesting level. You can put multiple context managers on the same `with` by comma-separating. But also, let's not use dynamo_timed just to get another waitcounter. We have this pattern elsewhere in the codebase where we've been using a waitcounter context manager directly. Search for something like: `with _WaitCounter("blah").guard():`

masnesral replied:
Actually I don't know why we need a 2nd waitcounter? Maybe you can just supply the same name override here and below? I kind doubt we need "compile_async", "async_compile.precompile", and "async_compile.wait" separately?

jovianjaison replied:
Yes, used the same waitcounter_name_override for both of these.

masnesral commented on torch/_dynamo/output_graph.py:
```diff
@@ -1547,6 +1547,7 @@ def call_user_compiler(self, gm: fx.GraphModule) -> CompiledFn:
             "OutputGraph.call_user_compiler",
             phase_name="backend_compile",
             log_pt2_compile_event=True,
+            log_waitcounter=True,
```
I guess you'll want a name override here too.

masnesral replied:
"compile_aot_autograd" maybe? (I'm just trying to identify a pattern vs. the other waitcounters you're adding"

masnesral commented on torch/_inductor/async_compile.py:
```diff
@@ -444,6 +445,7 @@ def wait(self, scope: dict[str, Any]) -> None:
                 log_pt2_compile_event=True,
                 dynamo_compile_column_us="triton_compile_time_us",
                 log_waitcounter=True,
+                waitcounter_name_override="compile_async",
```
Maybe we should name these "compile_triton"??

jovianjaison replied:
updated.

masnesral commented:
Looks consistent between dynamo_compile and waitcounters now. Thanks.

bdhirsh commented on torch/_inductor/async_compile.py:
```diff
@@ -444,6 +445,7 @@ def wait(self, scope: dict[str, Any]) -> None:
                 log_pt2_compile_event=True,
                 dynamo_compile_column_us="triton_compile_time_us",
                 log_waitcounter=True,
+                waitcounter_name_override="complie_async_triton",
```
nit: `compile_async_triton`? (same for the one above)

jovianjaison replied:
Yes, we want the same waitcounter name to track both.

masnesral commented on torch/_inductor/async_compile.py:
```diff
@@ -359,6 +359,7 @@ def get_result() -> CachingAutotuner:
                 log_pt2_compile_event=True,
                 dynamo_compile_column_us="triton_compile_time_us",
                 log_waitcounter=True,
+                waitcounter_name_override="complie_async_triton",
```
@jovianjaison you have a typo (complie). But actually, this instance is not technically "async", whereas the one below is. I agree they should be named the same, but "compile_triton" is probably slightly more accurate.

ppanchalia replied:
@masnesral I think the intent is to capture the phase of the compile, not the whether it's actually async or not. The intention is to measure time spent in "async compile" so that's why the wait counter name `compile_async_triton`

ppanchalia replied:
@jovianjaison fix the typo

jovianjaison replied:
@ppanchalia should I stick with compile_triton then?

ppanchalia replied:
@jovianjaison use `compile_async_triton` to capture time spend in the "async_compile" phase

masnesral replied:
> @masnesral I think the intent is to capture the phase of the compile, not the whether it's actually async or not. The intention is to measure time spent in "async compile" so that's why the wait counter name compile_async_triton

I don't really have a strong opinion, but if that's your intent then the name might be even more confusing because I don't think we really have an "async compile" phase per se. If anything, this is part of the "inductor" phase, and specifically the portion of inductor that's compiling the Triton.

ppanchalia replied:
The intent here is to capture, the time spent in this "async" phase where we look at RE interactions and time we're spending there

ppanchalia reviewed (changes requested):
fix typo

jovianjaison commented:
@pytorchbot revert -m "needs a fix"

jovianjaison commented:
@pytorchbot revert -m "needs a fix" -c "ghfirst"

jovianjaison committed:
Log aot and idx waitcounters. (#152444)

Summary:

Added for create_aot_dispatcher_function, compile_fx_inner and compile_async.

Note:
Log wait counters flag is already set for: 
1. async_compile.precompile: https://www.internalfb.com/code/fbsource/[9a8d1108ad1b]/fbcode/caffe2/torch/_inductor/async_compile.py?lines=361
2. remote_fx_graph_cache_get: https://www.internalfb.com/code/fbsource/[9a8d1108ad1b]/fbcode/caffe2/torch/_inductor/remote_cache.py?lines=50
3. remote_fx_graph_cache_put: https://www.internalfb.com/code/fbsource/[9a8d1108ad1b]/fbcode/caffe2/torch/_inductor/remote_cache.py?lines=58

Test Plan:
Compared:
dynamo_compile: https://fburl.com/daiquery/y58wygb1

vs

compile_inductor: https://pxl.cl/77cK4
compile_async: https://pxl.cl/77d5g

Reviewed By: ppanchalia

Differential Revision: D73866124

jovianjaison commented:
@pytorchbot merge
